src/editors/data/redux/thunkActions/app.js

```diff
@@ -44,9 +44,53 @@ export const saveBlock = ({ content, returnToUnit }) => (dispatch) => {
   }));
 };
 
-export const fetchImages = ({ onSuccess }) => () => {
+export const fetchImages = ({ onSuccess, stopLoading }) => () => {
   // get images
-  onSuccess(mockData.mockImageData);
+  const processedData = Array.from(mockData.mockImageData, el => {
+    const sortDate = new Date(el.dateAdded.replace(' at', '')).getTime();
+    return {
+      ...el,
+      sortDate,
+    };
+  });
+  stopLoading();
+  return onSuccess(processedData);
+};
+
+export const uploadImage = ({
+  file, startLoading, stopLoading, resetFile, setError,
+}) => () => {
+  // input file
+  // lastModified: 1643131112097
+  // lastModifiedDate: Tue Jan 25 2022 12:18:32 GMT-0500 (Eastern Standard Time) {}
+  // name: "Profile.jpg"
+  // size: 21015
+  // type: "image/jpeg"
+
+  // api will respond with the following JSON
+  // {
+  //   "asset": {
+  //     "display_name": "journey_escape.jpg",
+  //     "content_type": "image/jpeg",
+  //     "date_added": "Jan 05, 2022 at 21:26 UTC",
+  //     "url": "/asset-v1:edX+test101+2021_T1+type@asset+block@journey_escape.jpg",
+  //     "external_url": "https://courses.edx.org/asset-v1:edX+test101+2021_T1+type@asset+block@journey_escape.jpg",
+  //     "portable_url": "/static/journey_escape.jpg",
+  //     "thumbnail": "/asset-v1:edX+test101+2021_T1+type@thumbnail+block@journey_escape.jpg",
+  //     "locked": false,
+  //     "id": "asset-v1:edX+test101+2021_T1+type@asset+block@journey_escape.jpg"
+  //   },
+  //   "msg": "Upload completed"
+  // }
+
+  console.log(file);
+  startLoading();
+  setTimeout(() => {
+    stopLoading();
+    resetFile();
+    setError('test error');
```

Comment:
I currently have this coded to return an error for testing purposes.

Reply:
What is resetFile()?

I was simply going to do something like:

dispatch(requests.saveImage({
    image,
    onSuccess: (response) => onSuccess(response),
    onFailure: (e) => setError,
  }));

where onSuccess is a function which sets the loading state to done and sets the file to be selected. Would that break something here?